src/density/density.cpp

```diff
@@ -47,12 +47,12 @@ Density::Density(Simulation_context& ctx__)
 
     /*  allocate charge density and magnetization on a coarse grid */
     for (int i = 0; i < ctx_.num_mag_dims() + 1; i++) {
-        rho_mag_coarse_[i] = std::unique_ptr<spf>(new spf(ctx_.spfft_coarse(), ctx_.gvec_coarse_partition()));
+        rho_mag_coarse_[i] = std::unique_ptr<spf>(new spf(ctx_.spfft_coarse<double>(), ctx_.gvec_coarse_partition()));
```

Comment:
that's an important spot: in fp32 we will sum fp32 wave-functions squared into fp32 OR fp64 density depending on the accuracy considerations

Reply:
Would it sound more complete to introduce the required change in this PR? Looks like it would either require to template the class and function, or include many more variables to the class

Reply:
We will need to have summation of charge density in FP64 from the FP32 wave-functions to understand the accuracy. So let's try to keep Density and Potential in FP64.